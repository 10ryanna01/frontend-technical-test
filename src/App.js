import React from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { HelmetProvider } from 'react-helmet-async';
import VehicleList from './components/VehicleList';
import Layout from './components/Layout/Layout';
import SEO from './components/SEO/SEO';

export default function App() {
  const [queryClient] = React.useState(() => new QueryClient());
  const helmetContext = {};
  return (
    <main data-testid="appTest1">
      <QueryClientProvider client={queryClient} contextSharing>
        <HelmetProvider context={helmetContext}>
          <SEO
            title="APIs & Fetch Test Tech test"
            description="Sustainable chicharrones yuccie marfa pinterest kickstarter heirloom coloring book skateboard."
            name="Hammock Pitchfork"
            type="article"
          />
          <Layout>
            <VehicleList />
          </Layout>
        </HelmetProvider>
      </QueryClientProvider>
    </main>
  );
}
