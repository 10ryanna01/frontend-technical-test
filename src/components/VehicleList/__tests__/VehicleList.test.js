import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import App from '../../../App';

it('should render App component', async () => {
  render(<App />);

  const layoutTest1 = screen.getByTestId('appTest1');
  expect(layoutTest1).toBeInTheDocument();
});

it('capture a snap shot of App component', async () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});
