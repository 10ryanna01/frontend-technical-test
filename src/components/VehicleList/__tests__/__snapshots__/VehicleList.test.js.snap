// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`capture a snap shot of App  1`] = `
<main
  data-testid="appTest1"
>
  <div
    data-testid="seoTest1"
  />
  <div
    data-testid="layoutTest1"
  >
    loading...
  </div>
</main>
`;

exports[`capture a snap shot of App component  1`] = `
<main
  data-testid="appTest1"
>
  <div
    data-testid="seoTest1"
  />
  <div
    data-testid="layoutTest1"
  >
    loading...
  </div>
</main>
`;

exports[`capture a snap shot of App component 1`] = `
<main
  data-testid="appTest1"
>
  <div
    data-testid="seoTest1"
    role="site identity information"
  />
  <main
    className="UI__main-layout"
    data-testid="layoutTest1"
    role="site layout"
  >
    loading...
  </main>
</main>
`;

exports[`renders App correctly  1`] = `
<div
  data-testid="appTest1"
>
  <div
    data-testid="seoTest1"
  />
  <div
    data-testid="layoutTest1"
  >
    loading...
  </div>
</div>
`;

exports[`renders App correctly 1`] = `
<div
  data-testid="appTest1"
>
  loading...
</div>
`;

exports[`renders Vehicle correctly 1`] = `
<div
  data-testid="layoutTest1"
/>
`;
